REST API added for user to organization relationship

jekcosty opened the pull request:
This is being added to enable fetching of the organization related to a non-profit user

jekcosty committed:
Merge remote-tracking branch 'Code4SocialGood/master'

jekcosty committed:
Merge remote-tracking branch 'Code4SocialGood/master'

jekcosty committed:
REST API for user to org service

These are changes to add a REST API service to get user to organization
relationships.

jekcosty committed:
Merge branch 'Auth0-rel1.0.5'

felipealvesgnu commented on src/main/java/org/c4sg/mapper/OrganizationMapper.java:
```diff
@@ -17,4 +22,11 @@ public Organization getOrganizationEntityFromDto(OrganizationDTO organizationDTO
 		return organization;
 	}
+	
+	public OrganizationDTO getOrganizationDtoFromEntity(UserOrganization userOrganization){
```
Instead of create this method, what about reuse this method: 
`public OrganizationDTO getOrganizationDtoFromEntity(Organization organization){}`

and pass this `userOrganization.getOrganization()` as argument.

jekcosty replied:
That's right. Happens when you just copy from an existing one that does exactly the same function. We shall reuse.

felipealvesgnu commented on src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java:
```diff
@@ -21,6 +33,12 @@
+
+    @Autowired
+    private UserOrganizationDAO userOrganizationDAO;
+    
```
Keep attention on those identations.

jekcosty replied:
Not sure what it was exactly but I'll just Auto format the source.

felipealvesgnu commented on src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java:
```diff
@@ -88,4 +106,21 @@ public String getLogoUploadPath(Integer organizationId) {
+		Collections.sort(userOrganizations, (p1, p2) -> {
+			int result = 0;
+			result = p1.getStatus().compareTo(p2.getStatus());
+			return result*-1;
```
Why are you multiplying by -1 ?

jekcosty replied:
You got to ask whoever implemented for UserProject as this was just a clone and since it passed review my assumption is this worked. The only thing I could think of is perhaps something about the order.

felipealvesgnu replied:
I didn't get your point !

jekcosty replied:
Lol! I just copied/cloned this from an existing program that is for the UserProject entity. Looking at the latest changes I think status has been removed which should not have been there in the first place. I think the developer was trying to reverse the result of the comparison meaning it is sorting it out reverse of the natural sorting order. And I don't know why???

felipealvesgnu replied:
I found this piece of code with -1, we need to refactor this in the future
```
  @Override
    public List<ProjectDTO> findByUser(Integer userId) {
        User user = userDAO.findById(userId);
        requireNonNull(user, "Invalid User Id");
        List<UserProject> userProjects = userProjectDAO.findByUserId(userId);
        Comparator<UserProject> projectComp = new Comparator<UserProject>() {
            @Override
            public int compare(UserProject o1, UserProject o2) {
                int result = 0;
                // result = o1.getStatus().compareTo(o2.getStatus());
                return result * -1;
            }
        };
        Collections.sort(userProjects, projectComp);
        List<ProjectDTO> projectDtos = new ArrayList<ProjectDTO>();
        for (UserProject userProject : userProjects) {
            projectDtos.add(projectMapper.getProjectDtoFromEntity(userProject));
        }
        return projectDtos;
    }
```

felipealvesgnu commented on src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java:
```diff
@@ -88,4 +106,21 @@ public String getLogoUploadPath(Integer organizationId) {
+		User user = userDAO.findById(userId);
+        requireNonNull(user, "Invalid User Id");
+		List<UserOrganization> userOrganizations = userOrganizationDAO.findByUserId(userId);
+		Collections.sort(userOrganizations, (p1, p2) -> {
```
Try to use this in the Lambda expression it will be more readable 
`Comparator.comparing( [add the method here] )`

jekcosty replied:
It just so happens that Comparator.comparing is introduced in Java8 as a convenience function for say readability purpose, however, one can make an argument that it is a choice. Does the plain lambda expression suffer from performance issues? Well, not really right. If you check out most codes in the front-end, you would be shaking your head 1000 times. It is just a choice.

felipealvesgnu replied:
ok

felipealvesgnu reviewed (changes requested):
Please make those modifications

jekcosty committed:
Merge remote-tracking branch 'Code4SocialGood/master'

# Conflicts:
#	src/main/java/org/c4sg/controller/OrganizationController.java
#	src/main/java/org/c4sg/entity/UserOrganization.java
#	src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java

jekcosty committed:
Merge branch 'master' into Auth0-rel1.0.5

jekcosty committed:
Added recommended changes and fix for broken user entity mappings

Made changes as recommended for PR review on getting user to org
relationships. This also fixes what was broken when the user table was
altered with new columns added.

jekcosty committed:
Merge branch 'Auth0-rel1.0.5'

jekcosty committed:
Merge remote-tracking branch 'Code4SocialGood/master'

jekcosty committed:
Merge branch 'master' into Auth0-rel1.0.5

jekcosty commented:
Ok Felipe @felipealves-gnu , I've made some changes, please review. On some classes I've commented out codes for now rather than deleting them but the plan is Jenny and I and even you , we'll have to do some code sweeping/cleanup sometime before initial release.

felipealvesgnu committed:
Merge branch 'master' into master

jekcosty committed:
Merge remote-tracking branch 'Code4SocialGood/master'

# Conflicts:
#	src/main/java/org/c4sg/controller/OrganizationController.java
#	src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java

jekcosty committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/org/c4sg/controller/OrganizationController.java
#	src/main/java/org/c4sg/service/impl/OrganizationServiceImpl.java

jekcosty committed:
Merge branch 'master' into Auth0-rel1.0.5

jekcosty committed:
more corrections

jekcosty committed:
Merge branch 'Auth0-rel1.0.5'
